prog/mutation.go

```diff
@@ -183,6 +189,54 @@ func (ctx *mutator) mutateArg() bool {
 	return true
 }
 
+// Select a random argument weighted by its priority
+func (ma *mutationArgs) pseudoRandomArgs(r *randGen) (arg Arg, ctx ArgCtx) {
+	// map used to store the indexes of the arguments with the same priority
+	m := make(map[float64][]int)
+
+	sum := float64(0)
+	for _, value := range ma.priorities {
```

Comment:
Unit-test this code as well. The tested function should accept the priorities array (e.g. `[]float64{0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.2}`) and return an index in that array. Such input data is both simpler to create in the unit test and we will test more code.